code/game/gamemodes/clock_cult/clock_structures.dm

```diff
@@ -0,0 +1,755 @@
+	"<span class='heavy_brass'>The tendrils around your arm turn a horrible black and sting your skin before they shrivel away.</span>")
+	user.notransform = FALSE
+	if(!src)
+		return 0
+	flick("[initial(icon_state)]_discharged", src)
+	icon_state = "[initial(icon_state)]_recharging"
+	spawn(3000) //5 minutes
+		if(!src)
+			return 0
+		visible_message("<span class='warning'>The writhing tendrils return to the gemstone, which begins to glow with power.</span>")
+		flick("[initial(icon_state)]_recharged", src)
+		icon_state = initial(icon_state)
+		recharging = FALSE
+	return 1
+
+/obj/structure/clockwork/mending_motor //Mending motor: A prism that consumes replicant alloy to repair nearby mechanical servants at a quick rate.
+	name = "mending motor"
+	desc = "A dark onyx prism, held in midair by spiraling tendrils of stone."
+	clockwork_desc = "A powerful prism that rapidly repairs nearby mechanical servants and clockwork structures."
+	icon_state = "mending_motor"
+	construction_value = 20
+	break_message = "<span class='warning'>The prism collapses with a heavy thud!</span>"
+	var/stored_alloy = 0
+	var/max_alloy = 150
+	var/uses_alloy = TRUE
+	var/active = TRUE
+
+/obj/structure/clockwork/mending_motor/prefilled
+	stored_alloy = 30
+
+/obj/structure/clockwork/mending_motor/New()
+	..()
+	SSobj.processing += src
+	toggle() //Toggles off as soon as it's created, but starts online for reasons
+
+/obj/structure/clockwork/mending_motor/Destroy()
+	SSobj.processing -= src
+	..()
+
+/obj/structure/clockwork/mending_motor/examine(mob/user)
+	..()
+	if(is_servant_of_ratvar(user) || isobserver(user))
+		user << "It has [stored_alloy]/[max_alloy] units of replicant alloy."
+
+/obj/structure/clockwork/mending_motor/process()
+	if(!active)
+		return 0
+	if(!stored_alloy && uses_alloy)
+		visible_message("<span class='warning'>[src] emits an airy chuckling sound and falls dark!</span>")
+		toggle()
+		return 0
+	for(var/mob/living/simple_animal/hostile/anima_fragment/F in range(5, src))
+		if(F.health == F.maxHealth || F.stat)
+			continue
+		F.adjustBruteLoss(-15)
+		if(uses_alloy)
+			stored_alloy = max(0, stored_alloy - 2)
+	for(var/mob/living/simple_animal/hostile/clockwork_marauder/M in range(5, src))
+		if(M.health == M.maxHealth || M.stat)
+			continue
+		M.adjustBruteLoss(-M.maxHealth) //Instant because marauders don't usually take health damage
+		M.fatigue = max(0, M.fatigue - 15)
+		if(uses_alloy)
+			stored_alloy = max(0, stored_alloy - 2)
+	for(var/mob/living/silicon/S in range(5, src))
+		if(S.health == S.maxHealth || S.stat == DEAD || !is_servant_of_ratvar(S))
+			continue
+		S.adjustBruteLoss(-25)
+		S.adjustFireLoss(-25)
+		if(uses_alloy)
+			stored_alloy = max(0, stored_alloy - 5) //Much higher cost because silicons are much more useful
+	for(var/obj/structure/clockwork/C in range(5, src))
+		if(C.health == C.max_health)
+			continue
+		C.health = min(C.health + 10, C.max_health)
+		if(uses_alloy)
+			stored_alloy = max(0, stored_alloy - 1)
```

Comment:
Instead of checking everything in range(5, src) FOUR TIMES, you could search for atom/movable in that range just once and then check if(istype(anima) else if(marauder) else if(silicon)...